### PR TITLE
Implement PageTransporter and WebElementAction

### PR DESCRIPTION
**Pull request by JessickaMoya**

### CODE REVIEW

File: src/main/java/core/WebDriverManagerSingleton.java [hunk 1]
```diff
@@ -1,48 +1,104 @@
...
+    /**
+     * Gets the WebDriverWait.
+     *
+     * @return the WebDriverWait.
+     */
+    public WebDriverWait getWebDriverWait() {
+        return webDriverWait;
+    }
+
+    public void closeDriver() {
```
Comment (danielaSantaCruz) [inline src/main/java/core/WebDriverManagerSingleton.java:99]: Java doc is missing is some methods of this class

File: src/main/java/salesforce/ui/PageTransporter.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import core.WebDriverManagerSingleton;
+import salesforce.ui.pages.ProductPage;
+import utils.SetUp;
+
+/**
+ * Controls navigation of pages.
+ */
+public class PageTransporter {
+
+    public void goToUrl(final String url) {
```
Comment (danielaSantaCruz) [inline src/main/java/salesforce/ui/PageTransporter.java:18]: Java doc is missing is some methods of this class

File: src/main/java/salesforce/ui/pages/BasePage.java [hunk 3]
```diff
@@ -1,25 +1,26 @@
...
 
     /**
      * Constructor method.
-     *
-     * @param driver with the driver needed.
      */
-    public BasePage(final WebDriver driver) {
-        this.driver = driver;
-        wait = new WebDriverWait(driver, Long.valueOf(SetUp.WAIT_TIME_VALUE.getValue()));
+    public BasePage() {
```
Comment (JorgeRCV) [inline src/main/java/salesforce/ui/pages/BasePage.java:20]: If a driver is instanciated here, everytime a new page is clicked a new driver will bw instanciated and open a new browser

File: src/test/java/products/ProductsTests.java [hunk 4]
```diff
@@ -2,13 +2,18 @@
...
 import org.testng.annotations.Test;
+import salesforce.ui.pages.HomePage;
+import salesforce.ui.pages.LoginPage;
 import salesforce.ui.pages.ProductPage;
 import utils.SetUp;
 
 public class ProductsTests extends BaseTest {
 
-//    @Test
+    @Test
```
Comment (JorgeRCV) [inline src/test/java/products/ProductsTests.java]: The test does not assert anything

File: src/main/java/salesforce/ui/pages/BasePage.java (hunk 3 above)
Comment (JessickaMoya) [reply src/main/java/salesforce/ui/pages/BasePage.java:20]: Thanks, I have fixed that

File: src/test/java/products/ProductsTests.java (hunk 4 above)
Comment (JessickaMoya) [reply src/test/java/products/ProductsTests.java]: Thanks, I have fixed that

File: src/main/java/salesforce/ui/PageTransporter.java (hunk 2 above)
Comment (JessickaMoya) [reply src/main/java/salesforce/ui/PageTransporter.java:18]: Thanks, I have added that

File: src/main/java/core/WebDriverManagerSingleton.java (hunk 1 above)
Comment (JessickaMoya) [reply src/main/java/core/WebDriverManagerSingleton.java:99]: Thanks, I have added that

File: src/main/java/salesforce/ui/pages/HomePage.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package salesforce.ui.pages;
+
+/**
+ * This class returns an instance of HomePage.
+ */
+public class HomePage extends BasePage {
+
+    @Override
+    protected void waitForPageLoaded() {
+
```
Comment (JorgeRCV) [inline src/main/java/salesforce/ui/pages/HomePage.java:10]: This method is not implemented

**Review by JorgeRCV — APPROVED** [review]
Some checkstyle corrections are missing, after that everthing well done good job 🥇

**Review by danielaSantaCruz — COMMENTED** [review]
Good job! but all checks have failed 😅

**Review by danielaSantaCruz — APPROVED** [review]

**Review by danielaSantaCruz — APPROVED** [review]

**Pull request merged**